h3/src/proto/frame.rs

```diff
@@ -0,0 +1,543 @@
+use std::fmt;
+
+use bytes::{buf::ext::BufExt as _, Buf, BufMut, Bytes};
+use tracing::trace;
+
+use super::varint::{BufExt, BufMutExt, UnexpectedEnd, VarInt};
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    Malformed,
+    UnsupportedFrame(u64), // Known frames that should generate an error
+    UnknownFrame(u64),     // Unknown frames that should be ignored
+    UnexpectedEnd,
+    InvalidFrameValue,
+    InvalidSettingId(u64),
+    InvalidSettingValue(SettingId, u64),
+    SettingRepeated(SettingId),
+    SettingsExceeded,
+    Incomplete(usize),
+    IncompleteData,
+    Settings(String),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Frame {
```

Comment:
Just a suggestion: perhaps we can remove the `Frame` suffix from the inner types (so, `Headers` instead of `HeadersFrame`, etc). Since we're already in the `frame` module, I think it will still be clear.

Reply:
Done